Assets/SEE/UI/DebugAdapterProtocol/DebugAdapterProtocolSession.cs

```diff
@@ -0,0 +1,1233 @@
+        {
+            WindowSpace manager = WindowSpaceManager.ManagerInstance[WindowSpaceManager.LocalPlayer];
+            ConsoleWindow console = manager.Windows.OfType<ConsoleWindow>().FirstOrDefault();
+            if (console == null)
+            {
+                console = Canvas.AddComponent<ConsoleWindow>();
+                ConsoleWindow.DefaultChannel = "Adapter";
+                ConsoleWindow.DefaultChannelLevel = "Log";
+                manager.AddWindow(console);
+                if (start)
+                {
+                    foreach ((string channel, char icon) in new[] { ("Program", '\uf135'), ("Adapter", '\uf188') })
+                    {
+                        ConsoleWindow.AddChannel(channel, icon);
+                        foreach ((string level, Color color) in new[] { ("Log", Color.gray), ("Warning", Color.yellow.Darker()), ("Error", Color.red.Darker()) })
+                        {
+                            ConsoleWindow.AddChannelLevel(channel, level, color);
+                        }
+                    }
+                    ConsoleWindow.SetChannelLevelEnabled("Adapter", "Log", false);
+                }
+            }
+            manager.ActiveWindow = console;
+        }
+
+        /// <summary>
+        /// Opens the variables window.
+        /// Can be recalled to reopen or focus the console.
+        /// </summary>
+        private void OpenVariables()
+        {
+            WindowSpace manager = WindowSpaceManager.ManagerInstance[WindowSpaceManager.LocalPlayer];
+            if (variablesWindow == null)
+            {
+                variablesWindow = manager.Windows.OfType<VariablesWindow>().FirstOrDefault() ?? Canvas.AddComponent<VariablesWindow>();
+                variablesWindow.Variables = variables;
+                variablesWindow.RetrieveNestedVariables = RetrieveNestedVariables;
+                variablesWindow.RetrieveVariableValue = RetrieveVariableValue;
+                manager.AddWindow(variablesWindow);
+            }
+
+            manager.ActiveWindow = variablesWindow;
+        }
+
+
+        /// <summary>
+        /// Creates the process for the debug adapter.
+        /// </summary>
+        /// <returns>Whether the creation was sucessful.</returns>
+        private bool CreateAdapterProcess()
+        {
+            adapterProcess = new Process()
+            {
+                StartInfo = new ProcessStartInfo()
+                {
+                    FileName = Adapter.AdapterFileName,
+                    Arguments = Adapter.AdapterArguments,
+                    WorkingDirectory = Adapter.AdapterWorkingDirectory,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    RedirectStandardInput = true,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    // message headers are ASCII, message bodies are UTF-8
```

Comment:
As I understand it, UTF-8 includes everything from ASCII (thus backwards compatible) and even more. 
So if I set the encoding to UTF-8, characters that are not ASCII could be sent in the header. (which isn't allowed)

Reply:
Wouldn't the message bodies' UTF-8 encoding then lead to problems, since the process can only handle ASCII values? If DAP headers are always ASCII, changing the encoding here to a backwards-compatible format shouldn't be an issue, right? I'm unsure though—the best way to be sure is probably by testing this on some non-ASCII values (e.g., printing a string containing German umlauts using the DAP console).

Reply:
Sending non-ASCII values (tried German umlauts) with the ASCII-encoding seems to be working fine.
Sending any message with the UTF8-encoding causes to errors when trying to read the message.